### PR TITLE
Add a pybind function to clear a list.

### PR DESCRIPTION
**Pull request by TCKnet**
## Description

Add `py::list::clear()`.

See also: https://stackoverflow.com/questions/23489177/how-to-clear-a-pylistobject

## Suggested changelog entry:

```rst
``py::list`` gained a ``.clear()`` method.
```

### CODE REVIEW

**Comment by TCKnet** [comment]
It seems that this is a system/test configuration/version  issue rather than an issue with the CL per se. Could a maintainer have a look? Tthat would be awesome! Thank you very much in advance!

Using cmake version 3.29.4
Error: Could not find linux asset for cmake version 3.29.4
Using cmake version 3.29.4
Error: Could not find darwin asset for cmake version 3.29.4

**Comment by rwgk** [comment]
About to close/reopen this PR to trigger a fresh rerun of the GitHub Actions.

**Review by rwgk — APPROVED** [review]
LGTM. I think this is a useful small addition, especially because the implementation isn't exactly obvious (why is there no `PyList_Clear()`?).

File: include/pybind11/pytypes.h [hunk 1]
```diff
@@ -2183,6 +2183,11 @@ class list : public object {
             throw error_already_set();
         }
     }
+    void clear() {
```
Comment (Skylion007) [inline include/pybind11/pytypes.h:2186]: Can you add the non-const comment the other pybind11 clear methods have?

**Review by Skylion007 — APPROVED** [review]
Except for the comment issue, looks good.

File: include/pybind11/pytypes.h (hunk 1 above)
Comment (rwgk) [reply include/pybind11/pytypes.h]: Done. Thanks for catching this!

**Pull request merged**